frontend/lib/src/components/auth/auth_header.dart

```diff
@@ -3,26 +3,67 @@ import 'package:flutter/material.dart';
 /*
   Reusable "Top" component for auth related actions
   such as login, forgot password, create password
+
+  @param hasAuthToken, bool value to tell if user is authenticated or not.
 */
 class AuthHeader extends StatelessWidget {
-  const AuthHeader({super.key});
+  const AuthHeader({super.key, this.hasAuthToken = false});
```

Comment:
refactor so it would use the same technique as master branch,
accepting a child widget so we can have this as topmost widget

Reply:
@jstephend-sun 
I just put the AuthHeader on each screen navigator that will return AuthHeader and accept its child, this is okay for me cause nested pages can still inherit the header. If I will do the same technique as master branch, I will get an error constraints.biggest.isFinite I think because some other pages the whole page is scrollable it can't determine the exact height of the entire page and I have trouble finding solution for this error.  

For the login and password reset page, I just put directly header inside the page, I can't do the same technique in master where you get the page base on the index.